server/text_generation_server/layers/moe/__init__.py

```diff
@@ -152,6 +158,17 @@ def forward(self, x: torch.Tensor, *, gating_output: torch.Tensor) -> torch.Tens
         input_shape = x.shape
         x = x.view(-1, input_shape[-1])
 
+        if SYSTEM == "ipex":
+            return self.ipex_fused_moe(
+                hidden_states=x,
+                router_logits=gating_output,
+                top_k=self.topk,
+                renormalize=self.renormalize,
+                use_grouped_topk=self.n_expert_group is not None,
+                num_expert_group=self.n_expert_group,
+                topk_group=self.topk_group,
+            )
+
```

Comment:
I am not sure if we want this here, the idea of this class is to provide the a non-fused implementation. It's a fallback for when no fused implementation is available (e.g. for platforms where no fused kernels are available for GPTQ/AWQ).

Reply:
from moe_kernels.fused_moe import fused_topk, grouped_topk,  is moe_kernel only for cuda?

Reply:
Ah, yes, I see the issue now. Thanks!